subsys/bluetooth/host/l2cap.c

```diff
@@ -582,14 +582,25 @@ int bt_l2cap_send_cb(struct bt_conn *conn, uint16_t cid, struct net_buf *buf,
 		     bt_conn_tx_cb_t cb, void *user_data)
 {
 	struct bt_l2cap_hdr *hdr;
+	struct net_buf_simple_state state;
+	int err;
 
 	BT_DBG("conn %p cid %u len %zu", conn, cid, net_buf_frags_len(buf));
 
+	net_buf_simple_save(&buf->b, &state);
+
 	hdr = net_buf_push(buf, sizeof(*hdr));
 	hdr->len = sys_cpu_to_le16(buf->len - sizeof(*hdr));
 	hdr->cid = sys_cpu_to_le16(cid);
 
-	return bt_conn_send_cb(conn, buf, cb, user_data);
+	err = bt_conn_send_cb(conn, buf, cb, user_data);
+	if (!err) {
+		return 0;
+	}
+
+	net_buf_simple_restore(&buf->b, &state);
```

Comment:
Buffer is most probably freed at this point with `net_buf_unref(buf);` inside `bt_conn_send_cb()`. So any operation at buffer is undefined in results.

Reply:
Just fixed that.